forc-pkg/src/pkg.rs

```diff
@@ -454,6 +485,20 @@ fn validate_dep_manifest(dep: &Pinned, dep_manifest: &ManifestFile) -> Result<()
             dep_manifest.project.name,
         );
     }
+    if let Some(dep_forc_version) = &dep_manifest.project.forc_version {
+        let crate_version = env!("CARGO_PKG_VERSION");
+        // Ensure the current forc version is >= forc_version required for this dep
+        let current_forc_version = semver::Version::parse(crate_version)?;
+        if current_forc_version < *dep_forc_version {
+            bail!(
+                "{:?} requires forc version {} but current forc version is {}\nUpdate the toolchain by following: https://fuellabs.github.io/sway/v{}/introduction/installation.html",
+                dep.name,
+                dep_forc_version,
+                current_forc_version,
+                crate_version
+            );
+        }
+    }
```

Comment:
It looks like this whole `if let` block could be replaced with your new function? I.e.
```rust
    validate_version(dep_manifest)?;
```

Reply:
Ahh yeah my bad. Thanks for pointing it!